mllib/src/main/scala/org/apache/spark/mllib/clustering/KMeans.scala

```diff
@@ -193,20 +210,29 @@ class KMeans private (
 
     val initStartTime = System.nanoTime()
 
-    val centers = if (initializationMode == KMeans.RANDOM) {
-      initRandom(data)
-    } else {
-      initKMeansParallel(data)
-    }
+    val numRuns = if (initialModel.nonEmpty) runs else 1
 
+    val centers = initialModel match {
+      case Some(kMeansCenters) => {
+        Array.tabulate(numRuns)(r => kMeansCenters.clusterCenters
```

Comment:
No need to use numRuns or Array.tabulate here since you know there is only 1 run